### PR TITLE
fix(Datagrid): only remove single filter for correct table id

### LINKED ISSUE #4522: DataGrid Filter issues
## What package(s) are you using?

- [ ] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive). Yes

## Detailed description

1.   Create DataGrid along with Filter.
2.   Add a filter of type Checkbox.
3.   Add checkbox filter details as shown in the notes section.
4.   Select the checkbox filter from the filter flyout  and click on Apply.
5.   Now try to remove the filter with label Glass using the X icon  , this fails. ( As per my observation this happens when using different value for id and value attribute of the checkbox filter)

> Is this issue related to a specific component?
DataGrid Filter

> What did you expect to happen? What happened instead? What would you like to
On click of X , it should remove the selected filter.

> What browser are you working in?
Chrome and safari

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?
"@carbon/ibm-products": "2.30.0",
 "@carbon/react": "1.52.0",

> Please create a reduced test case in CodeSandbox
https://codesandbox.io/p/sandbox/brave-poincare-2ffcfr?file=%2Fpackage.json%3A56%2C2

- Notes
  {
    type: "checkbox",
    column: "lastName",
    props: {
      FormGroup: {
        legendText: "Last Name",
      },
      Checkbox: [
        {
          id: "glass1",
          labelText: "Glass",
          value: "glass",
        },
        {
          id: "regret",
          labelText: "Regret",
          value: "regret",
        },
        {
          id: "current",
          labelText: "Current",
          value: "current",
        },
      ],
    },
  },

### PR DESCRIPTION
**Pull request by matthewgallo**
Closes #4522 

This fixes a bug where removing a filter tag when there are multiple instances of a Datagrid component it gets removed for both Datagrids. With this change, the filter tag will only be removed from the appropriate Datagrid.

This is similar to the fix that @annawen1 made with the "Clear filters" button but for the removal of single filter tags.

#### What did you change?
```
packages/ibm-products/src/components/Datagrid/Datagrid/DatagridContent.tsx
packages/ibm-products/src/components/Datagrid/Datagrid/addons/Filtering/FilterProvider.js
```
#### How did you test and verify your work?
Storybook, temporarily created a new story that used `Tabs` to render two separate Datagrid components.

### CODE REVIEW

File: packages/ibm-products/src/components/Datagrid/Extensions/Filtering/Panel.stories.jsx [hunk 1]
```diff
@@ -24,7 +26,7 @@ export default {
   parameters: {
     styles,
     docs: { page: DocsPage },
-    layout: 'fullscreen',
+    layout: 'padded',
```
Comment (elycheea) [inline packages/ibm-products/src/components/Datagrid/Extensions/Filtering/Panel.stories.jsx:29]: Guessing we want to keep these `fullscreen`?

Comment (matthewgallo) [reply packages/ibm-products/src/components/Datagrid/Extensions/Filtering/Panel.stories.jsx:29]: Yep, that was leftover from the tab example. I'll revert to `fullscreen`.

**Review by elycheea — APPROVED** [review]
😆

**Review by AlexanderMelox — APPROVED** [review]
This looks good to me 👍🏻

**Pull request merged**